mmv1/products/vertexai/IndexEndpoint.yaml

```diff
@@ -97,3 +102,12 @@ properties:
       [Format](https://cloud.google.com/compute/docs/reference/rest/v1/networks/insert): `projects/{project}/global/networks/{network}`.
       Where `{project}` is a project number, as in `12345`, and `{network}` is network name.
     immutable: true
+  - !ruby/object:Api::Type::Boolean
+    name: publicEndpointEnabled
+    immutable: true
+    ignore_read: true
```

Comment:
[MEMO]

I confirmed a response of [projects.locations.indexEndpoints.get](https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/projects.locations.indexEndpoints/get) didn't include the publicEndpointEnabled like below

<details><summary>Details</summary>


```console
$ curl -X GET -H "Authorization: Bearer $(gcloud auth print-access-token)" https://us-central1-aiplatform.googleapis.com/v1/projects/xxx/locations/us-central1/indexEndpoints/5155064664874287104
{
  "name": "projects/xxxxx/locations/us-central1/indexEndpoints/5155064664874287104",
  "displayName": "sample-endpoint",
  "description": "A sample vertex endpoint with an public endpoint",
  "etag": "AMEw9yPkWvTjGqy1FSFN3Dc9NTn8J7Qaf09YteDcAF8JL6QaABLNasuBBOT1Cd9vTH4O",
  "labels": {
    "label-one": "value-one"
  },
  "createTime": "2023-09-02T19:29:32.917516Z",
  "updateTime": "2023-09-02T19:29:33.681500Z",
  "publicEndpointDomainName": "2086866499.us-central1-415528280434.vdb.vertexai.goog"
}
```

</details> 

In addition, I confirmed [projects.locations.indexEndpoints.patch](https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/projects.locations.indexEndpoints/patch) couldn't update the field like below:

<details><summary>Details</summary>

```console
$ curl -X PATCH -H "Authorization: Bearer $(gcloud auth print-access-token)" https://us-central1-aiplatform.googleapis.com/v1/projects/xxxx/locations/us-central1/indexEndpoints/5155064664874287104\?updateMask\=publicEndpointEnabled -H 'Content-Type: application/json' -d '{
 "description": "A sample vertex endpoint with an public endpoint",
 "displayName": "sample-endpoint",
 "labels": {
  "label-one": "value-one"
 },
 "publicEndpointEnabled": false
}'

{
  "name": "projects/xxxx/locations/us-central1/indexEndpoints/5155064664874287104",
  "displayName": "sample-endpoint",
  "description": "A sample vertex endpoint with an public endpoint",
  "labels": {
    "label-one": "value-one"
  },
  "createTime": "2023-09-02T19:29:32.917516Z",
  "updateTime": "2023-09-02T19:37:58.913067Z",
  "publicEndpointDomainName": "2086866499.us-central1-415528280434.vdb.vertexai.goog"
}
```

</details>